src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
+    public MyLinkedList() {
+        this.size = 0;
+        this.firstNode = null;
+        this.lastNode = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, lastNode, null);
+        if (this.isEmpty()) {
+            this.firstNode = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        this.lastNode = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) { // add to end of list or to empty list (index == size == 0)
+            this.add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(value, null, firstNode);
+            this.firstNode.prev = newNode;
+            this.firstNode = newNode;
+            size++;
+        } else {
+            Node<T> nextNode = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(value, nextNode.prev, nextNode);
+            nextNode.prev = newNode;
+            newNode.prev.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> findedNode = findNodeByIndex(index);
+        T oldValue = findedNode.item;
+        findedNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> findedNode = findNodeByIndex(index);
+        T oldValue = findedNode.item;
+        unlinkNode(findedNode);
+        size--;
+        return oldValue;
     }
 
     @Override
     public boolean remove(T object) {
+        if (size == 0) {
+            return false;
+        }
+        for (Node<T> currentNode = firstNode;
+                     currentNode != null;
+                     currentNode = currentNode.next) {
```

Comment:
```suggestion
        for (Node<T> currentNode = firstNode;
                     currentNode != null;
                     currentNode = currentNode.next) {
```
it will work perfectly